sites/cheerpx/src/content/docs/00-overview.mdx

```diff
@@ -54,23 +54,23 @@ WebContainers [can only execute languages that are natively supported on the Web
 
 ### CheerpX versus porting to WebAssembly
 
-[Porting a large C/C++ codebase to WebAssembly](/cheerp) is not necessarily a simple endeavor. For instance, a programming language implementation will most likely have multiple target-dependent code paths or definition, so the effort required is not dissimilar from porting the language to a new native architecture.
+[Porting a large C/C++ codebase to WebAssembly](/cheerp) is not necessarily a simple endeavor. For instance, a complex application will most likely have multiple target-dependent code paths or definition, so the effort required is not dissimilar from porting the application to a new native architecture.
 
 ### CheerpX versus v86/JSLinux
 
-CheerpX virtualises at a higher level. v86 and JSLinux emulate specific hardware peripherals, whereas CheerpX implements Linux system calls.
+CheerpX virtualization the system as a higher level. While v86 and JSLinux emulate specific hardware peripherals, CheerpX implements Linux system calls.
 
-They also require the entire disk image to be downloaded ahead-of-time (so [full 600MB Debian](https://webvm.io) is not really an option). CheerpX can stream the disk image from the server as needed.
+They also require the entire disk image to be downloaded ahead-of-time. A [2GB complete Debian system](https://webvm.io) would not be a practical option. CheerpX can stream the disk image from the server as needed.
 
-v86 and JSLinux are hobby projects, whereas CheerpX is a commercial product with a dedicated team of engineers. If you're shipping with CheerpX, you can [rely on us to provide support and maintenance](https://leaningtech.com/contact-us).
+Moreover, v86 and JSLinux are hobby projects, whereas CheerpX is a commercial product with a dedicated team of engineers. If you're shipping with CheerpX, you can [rely on us to provide support and maintenance](https://leaningtech.com/contact-us).
 
 ## How it works
 
-At its core, CheerpX is a two-tier emulator for the x86 architecture [implemented in C++ and compiled to JavaScript and WebAssembly](/cheerp). There is both an interpreter and a sophisticated JIT compiler that is able to generate efficient WebAssembly representations for hot code. The whole process is 100% automated and requires no metadata or assumptions whatsoever. The system is so robust that it can deal with applications that generate new code internally at runtime, and even with self-modifying code.
+At its core, CheerpX is a two-tier emulator for the x86 architecture [implemented in C++ and compiled to JavaScript and WebAssembly](/cheerp). There is both an interpreter and a sophisticated JIT compiler that is able to generate efficient WebAssembly representations for hot code. The whole process is 100% automated and requires no metadata or preprocessing steps. The system is so robust that it can deal with applications that generate new code internally at runtime, and even with self-modifying code.
```

Comment:
```suggestion
At its core, CheerpX is a two-tier emulator for the x86 architecture [implemented in C++ and compiled to JavaScript and WebAssembly](https://cheerp.io/). There is both an interpreter and a sophisticated JIT compiler that is able to generate efficient WebAssembly representations for hot code. The whole process is 100% automated and requires no metadata or preprocessing steps. The system is so robust that it can deal with applications that generate new code internally at runtime, and even with self-modifying code.
```